components/dropdown/script.js

```diff
@@ -795,29 +874,66 @@ class GamefaceDropdown extends CustomElementValidator {
 class DropdownOption extends HTMLElement {
     // eslint-disable-next-line require-jsdoc
     static get observedAttributes() {
-        return ['disabled'];
+        return ['disabled', 'selected', 'value'];
+    }
+
+    // eslint-disable-next-line require-jsdoc
+    get selected() {
+        return this.isAlreadySelected(this.closest('gameface-dropdown'), this) ||
+            this.hasAttribute('selected');
     }
 
     // eslint-disable-next-line require-jsdoc
     get value() {
-        return this.getAttribute('value') || this.textContent;
+        return this._value || this.textContent;
+    }
+
+    /**
+     * Checks if an option is already selected
+     * @param {HTMLElement} parent
+     * @param {HTMLElement} option
+     * @returns {boolean}
+     */
+    isAlreadySelected(parent, option) {
+        let isInParentsSelectedList = false;
+
+        for (const selectedOption of parent.selectedOptions) {
+            if (selectedOption !== option) continue;
+            isInParentsSelectedList = true;
+            break;
+        }
+
+        return isInParentsSelectedList;
     }
 
     /**
      * Called when an attribute changes
+     * @param {string} name
+     * @param {string} oldValue
+     * @param {string} newValue
+     * @returns {void}
     */
-    attributeChangedCallback() {
-        if (this.hasAttribute('disabled')) {
-            this.classList.add('guic-dropdown-option-disabled');
-        } else {
-            this.classList.remove('guic-dropdown-option-disabled');
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (name === 'disabled') {
+            if (this.hasAttribute('disabled')) return this.classList.add('guic-dropdown-option-disabled');
+            return this.classList.remove('guic-dropdown-option-disabled');
+        }
+
+        if (name === 'selected') {
+            const parent = this.closest('gameface-dropdown');
+            if (!parent || !parent.isRendered) return;
+            const hasSelected = newValue !== null;
+
+            if (hasSelected && parent.setSelectedAndScroll) return parent.setSelectedAndScroll(this, false, false);
+            if (!hasSelected && parent.deselect) parent.deselect(this, false);
```

Comment:
Same as the prev comment. Why do we check if the parent has `deselect` method? Is it possible the parent not to have it?

Reply:
Maybe comment these lines if it makes sense to check for the parent methods by describing why we do it.

Reply:
Yes, added a comment.